lib/matplotlib/axes/_base.py

```diff
@@ -2374,10 +2373,18 @@ def _update_patch_limits(self, patch):
                 ((not patch.get_width()) and (not patch.get_height()))):
             return
         p = patch.get_path()
-        vertices = p.vertices if p.codes is None else p.vertices[np.isin(
-            p.codes, (mpath.Path.CLOSEPOLY, mpath.Path.STOP), invert=True)]
-        if not vertices.size:
-            return
+        # Get all vertices on the path
+        # Loop through each sement to get extrema for Bezier curve sections
+        vertices = []
+        for curve, code in p.iter_bezier():
+            # Get distance along the curve of any extrema
+            _, dzeros = curve.axis_aligned_extrema()
+            # Calculate vertcies of start, end and any extrema in between
+            vertices.append(curve([0, *dzeros, 1]))
+
+        if len(vertices):
+            vertices = np.row_stack(vertices)
```

Comment:
Do you not want to return here if empty any more?

Reply:
Perhaps? It's a private helper function and doesn't seem to break anything though - I guess we're always passing in input that has some vertices.

Reply:
That's because `transform` and `update_datalim` are safe to empty lists. I think this is more of an optimization to avoid decomposing the transform, etc.